src/map/MapCeremonyPhases.jsx

```diff
@@ -39,6 +44,27 @@ export default React.memo(function MapCeremonyPhases (props) {
     return (small && currentPhase === 2) ? phasesProps.reverse() : phasesProps;
   };
 
+  const apiReady = (api, queryName = '') => {
+    const query = api && api.queryMulti && api.query;
+    return query && queryName ? (!!query[queryName]) : !!query;
+  };
+
+  const [nextMeetupTime, setNextMeetupTime] = useState([]);
+  // gets the date of the next Meetup
+  useEffect(() => {
+    if (!apiReady(api, 'encointerScheduler')) {
+      return;
+    }
+    async function getNextMeetupDate () {
+      const meetupLocations = await api.rpc.encointer.getLocations(cids[0]);
+      const tempLocation = locationFromJson(api, meetupLocations[0]);
+      const tempTime = await getNextMeetupTime(api, tempLocation);
+      debug && console.log('the date is' + formatDate(tempTime.toNumber()));
+      setNextMeetupTime(formatDate(tempTime.toNumber()).split(',')[0]);
+    }
+    getNextMeetupDate();
+  }, [api, cids, debug]);
```

Comment:
All of your `isMounted` hacks were only necessary because you introduced dependency cycles of your `setState` call.